### PR TITLE
r/storage_account: making `default_action` required

### LINKED ISSUE #4006: Omitting `default_action` should result in a deployment failure.
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform: 0.12.6
Provider: 1.32.1

### Affected Resource(s)

* `azurerm_storage_account`

### Terraform Configuration Files

```hcl
resource "azurerm_storage_account" "storage_account" {
  depends_on               = [""]

  name                     = ""
  location                 = ""
  resource_group_name      = ""
  account_tier             = "Standard"
  account_kind             = "StorageV2"
  account_replication_type = "RAGRS"

  network_rules {
    ip_rules                   = [""]
    virtual_network_subnet_ids = [""]
  }
}
```

### Expected Behavior

If the required attribute `default_action` within the `network_rules` section is omitted, Terraform should fail to deploy.

Initially we never used this attribute and our Azure Storage Accounts were not publically accessible. We only noticed recently that it's no longer the case. After reviewing the provider documentation we became aware of the required `default_action` attribute which we did not have, and although it's required, our deployments have been successful.

### Actual Behavior

Successful deployment. I assume that if it's omitted, it defaults to `Allow`.

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Fixes #4006

### CODE REVIEW

File: azurerm/resource_arm_storage_account.go [hunk 1]
```diff
@@ -205,7 +206,7 @@ func resourceArmStorageAccount() *schema.Resource {
 
 						"default_action": {
 							Type:     schema.TypeString,
-							Optional: true,
+							Required: true,
```
Comment (paddycarver) [inline azurerm/resource_arm_storage_account.go]: Is this not a breaking change?

Comment (katbyte) [reply azurerm/resource_arm_storage_account.go]: The title says "changing default to allow" and the change is to make it required? 🤔

Comment (tombuildsstuff) [reply azurerm/resource_arm_storage_account.go]: 🤦‍♂ my bad.

This is currently set to Required in the documentation: https://www.terraform.io/docs/providers/azurerm/r/storage_account.html#default_action - as such this PR updates it to match.. I'll update the title.

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2019-08-08 at 15 37 57

File: azurerm/resource_arm_storage_account.go (hunk 1 above)
Comment (paddycarver) [reply azurerm/resource_arm_storage_account.go]: I'll defer to you and @katbyte on whether you want to call this a breaking change or not.

Comment (tombuildsstuff) [reply azurerm/resource_arm_storage_account.go]: given this block is now Computed (and the API returns a default response here) I think we're  safe to not call this a breaking change here

**Comment by tombuildsstuff** [comment]
Tests passed:

Screenshot 2019-08-09 at 10 23 35

**Pull request merged**